postgrenosql/src/test/java/com/yahoo/ycsb/postgrenosql/PostgreNoSQLDBClientTest.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright 2017 YCSB Contributors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package com.yahoo.ycsb.db;
+
+import com.yahoo.ycsb.*;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.postgresql.Driver;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assume.assumeNoException;
+
+/**
+ * PostgreNoSQL test client for YCSB framework.
+ */
+public class PostgreNoSQLDBClientTest {
+  private static final Logger LOG = LoggerFactory.getLogger(PostgreNoSQLDBClientTest.class);
+
+  /** The default port for PostgreSQL. */
+  private static final int DEFAULT_PORT = 5432;
+  private static final String DATABASE_NAME = "test";
+  private static final String DEFAULT_USER = "postgres";
+  private static final String DEFAULT_PWD = "postgres";
+
+  /** The properties settings */
+  private static final String HOST_NAME  = "localhost";
+  private static final String TEST_DB_URL = "jdbc:postgresql://" + HOST_NAME + ":" + DEFAULT_PORT + "/" + DATABASE_NAME;
+  private static final String TABLE_NAME = "usertable";
+  private static final int FIELD_LENGTH = 32;
+  private static final String FIELD_PREFIX = "FIELD";
+  private static final int NUM_FIELDS = 3;
+
+  private static Connection postgreSQLConnection = null;
+  private static PostgreNoSQLDBClient postgreNoSQLClient = null;
+
+  @BeforeClass
+  public static void setUp() {
+    // Check whether postgres is available
+    try (Socket socket = new Socket(HOST_NAME, DEFAULT_PORT)){
+      assertThat("Socket is not bound.", socket.getLocalPort(), not(-1));
+    } catch (IOException connectFailed) {
+      assumeNoException("PostgreSQL is not running. Skipping tests.", connectFailed);
```

Comment:
excellent test addition!

Could you update our travis config to add a postgres instance? that'll make sure we get coverage through CI even if individual contributors don't have a local instance running.

https://docs.travis-ci.com/user/database-setup/#PostgreSQL